components/brave_rewards/resources/extension/brave_rewards/content_scripts/reddit.ts

```diff
@@ -97,6 +97,13 @@ const getRedditMetaData = (redditPost: Element): RewardsTip.MediaMetaData | null
     userName = anchor.textContent.startsWith('u/') ? anchor.textContent.split('/')[1] : anchor.textContent
   }
 
+  if (userName.length == 0) {
+    const spanUsername = redditPost.getElementsByClassName("lizQBHVukyun2S2babj-l _2tbHP6ZydRpjI44J3syuqC")[0]
+    if (spanUsername && spanUsername.textContent) {
+      userName = spanUsername.textContent.startsWith('u/') ? spanUsername.textContent.split('/')[1] : spanUsername.textContent
+    }
+  }
+
```

Comment:
I'm not entirely happy relying on that span's classname (given it's gobbledygook!) - ideas welcome if any.

However, if they change that classname, this will fail to get the userName - and the issue will re-occur. What would be the better UX? Hide the 'Tip' button altogether if no userName is available, or show some form of alert/error in the tip pane instead?

As it stands, nothing happens which is poor UX IMHO.

Reply:
@alexwykoff @jenn-rhim Hi there, can you have a think about my comment above and let us know what the preferred approach would be for this? TIA!